mangaki/mangaki/models.py

```diff
@@ -110,6 +115,29 @@ class Meta:
     def get_absolute_url(self):
         return reverse('work-detail', args=[self.category.slug, str(self.id)])
 
+    def retrieve_tags(self, anidb):
+        anidb_tags_list = anidb.get(self.anidb_aid).tags
+        anidb_tags = dict((tag[0], int(tag[1])) for tag in anidb_tags_list)
+
+        tag_work = TaggedWork.objects.filter(work=self)
+        current_tags = {tagwork.tag.title: tagwork.weight for tagwork in tag_work}
+
+        deleted_tags_keys = current_tags.keys() - anidb_tags.keys()
+        deleted_tags = dict((key, current_tags[key])for key in deleted_tags_keys)
```

Comment:
@fravoi Si tu ne trouves pas après avoir lu la doc de Python:
`dict((key, value) for blabla in blabla)` décomposé:

- `item for item in container` = generator expression. Cela fabrique un générateur (donc c'est pas complètement évalué en mémoire). (pense `for item in container: yield item`).
- `dict((key, value) for blabla in blabla)` = appelle le constructeur de dict sur un `iterable` (soit un générateur, soit une liste, etc…) de tuples `(key, value)` et construit un dictionnaire.

---

La version `{ key: value for item in container }` est une dict comprehension comme une list comprehension : `[ item for item in container ]`, le truc c'est qu'il me semble que le bytecode (le code généré par Python) pour cette syntaxe est _plus rapide_ plus ou moins.